msm/src/circuit_design/capabilities.rs

```diff
@@ -36,6 +36,10 @@ pub trait ColAccessCap<F: PrimeField, CIx: ColumnIndexer> {
     /// Turns a constant value into a variable.
     fn constant(value: F) -> Self::Variable;
 }
+/// Environment capability for accessing the next row, constraint only.
+pub trait NextCap<F: PrimeField, CIx: ColumnIndexer>: ColAccessCap<F, CIx> {
+    fn read_next(&self, col: CIx) -> Self::Variable;
```

Comment:
I'd strongly suggest to put this into `ColAccessCap` as `read_column_next`: it's literally the same semantics (reading a column). Any reasons why you didn't go for this design?